Triangulation_3/include/CGAL/internal/Triangulation_segment_traverser_3_impl.h

```diff
@@ -0,0 +1,1152 @@
+            o[li] = COPLANAR;
+            continue;
+        }
+
+        // Skip the previous cell.
+        Cell_handle next = cell()->neighbor(li);
+        if( next == prev_cell() ) {
+            o[li] = POSITIVE;
+            continue;
+        }
+
+        Point* backup = vert[li];
+        vert[li] = &(_target);
+        o[li] = _tr->orientation( *vert[0], *vert[1], *vert[2], *vert[3] );
+
+        if( o[li] != NEGATIVE ) {
+            vert[li] = backup;
+            continue;
+        }
+
+        // The target lies behind the plane through the source and two finite vertices.
+        // Traverse to the incident infinite cell.
+        CGAL_triangulation_assertion( _tr->is_infinite( next ) );
+        _prev = Simplex( cell(), Tr::FACET, li, -1 );
+        _cur = Simplex( next, Tr::FACET, next->index( prev_cell() ), -1 );
+        return;
+    }
+
+    // The line enters the convex hull here (or lies on the finite facet).
+    prev_cell() = cell();
+    cell() = fin;
+
+    // Check through which simplex the line traverses.
+    switch( o[0]+o[1]+o[2]+o[3] ) {
+        case 3:
+            prev_lt() = Tr::FACET;
+            prev_li() = inf;
+            lt() = Tr::FACET;
+            this->li() = cell()->index(prev_cell());
+            return;
+        case 2:
+            prev_lt() = Tr::EDGE;
+            lt() = Tr::EDGE;
+            for( int i = 0; i < 4; ++i ) {
+                if( o[i] == COPLANAR && i != inf ) {
+                    Edge opp = opposite_edge( prev_cell(), inf, i );
+                    prev_li() = opp.second;
+                    prev_lj() = opp.third;
+                    this->li() = cell()->index( prev_cell()->vertex( prev_li() ) );
+                    this->lj() = cell()->index( prev_cell()->vertex( prev_lj() ) );
+                    return;
+                }
+            }
+            CGAL_triangulation_assertion( false );
+            return;
+        case 1:
+            prev_lt() = Tr::VERTEX;
+            lt() = Tr::VERTEX;
+            for( int i = 0; i < 4; ++i ) {
+                if( o[i] == POSITIVE ) {
+                    prev_li() = i;
+                    this->li() = cell()->index( prev_cell()->vertex(i) );
+                    return;
+                }
+            }
+            CGAL_triangulation_assertion( false );
+            return;
+        default:
+            CGAL_triangulation_assertion( false );
+            return;
+    }
+}
+
+template < class Tr, class Inc >
+void Triangulation_segment_cell_iterator_3<Tr,Inc>::
+walk_to_next_2()
+{
+    std::array<Point*, 3> vert
+              = { &(cell()->vertex(0)->point()),
+                  &(cell()->vertex(1)->point()),
+                  &(cell()->vertex(2)->point()) };
+
+    switch( lt() ) {
+        case Tr::VERTEX: {
+            // First we try the incident edges.
+            Orientation ocw = typename Gt::Coplanar_orientation_3()( *vert[li()], *vert[_tr->cw(li())], *vert[_tr->ccw(li())], _target );
```

Comment:
I did a quick hack so that it compiles but your probably want to replace that by `tr.traits.coplanar_orientation_3_object()` or have that object stores. Same for `Collinear_3`.